### PR TITLE
Migrate 'sidebar permalink' e2e tests to Playwright

### PR DESCRIPTION
**Pull request by Mamaduka**
## What?
Part of #38851.
Supersedes and closes #42585.

PR migrates `sidebar-permalink.test.js` e2e test to Playwright

## Why?
See #38851.

## How?
The migration is based on #42585, which I think is abandoned. I've included the author so they get credit for their contribution.

## Testing Instructions
```
npm run test:e2e:playwright -- test/e2e/specs/editor/various/sidebar-permalink.spec.js
```

### CODE REVIEW

File: test/e2e/specs/editor/various/sidebar-permalink.spec.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	test( 'should not render URL when post is public but not publicly queryable', async ( {
+		admin,
+		editor,
+		page,
+	} ) => {
+		await admin.createNewPost( { postType: 'not_public_q_public' } );
+		await editor.canvas
+			.getByRole( 'textbox', { name: 'Add title' } )
+			.fill( 'aaaaa' );
+		await editor.saveDraft();
```
Comment (Mamaduka) [inline test/e2e/specs/editor/various/sidebar-permalink.spec.js:51]: The` editor.publishPost()` helper could fail for specific post-type configurations when the post-publish panel isn't displayed. I will double-check if not displaying the panel is intentional and fix the code accordingly.

**Failing locator**
https://github.com/WordPress/gutenberg/blob/22dcbde7c2f61d7534f262c3c38fe277063821ea/packages/e2e-test-utils-playwright/src/editor/publish-post.ts#L31-L34

File: test/e2e/specs/editor/various/sidebar-permalink.spec.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+			.getByRole( 'textbox', { name: 'Add title' } )
+			.fill( 'aaaaa' );
+		await editor.publishPost();
+		// Start editing again.
+		await editor.canvas
+			.getByRole( 'textbox', { name: 'Add title' } )
+			.fill( 'aaaa (Updated)' );
+		await expect(
+			page.getByRole( 'button', { name: 'Change URL' } )
+		).toBeHidden();
```
Comment (WunderBart) [inline test/e2e/specs/editor/various/sidebar-permalink.spec.js:38]: Should we ensure this button is visible before the post-publish editing?

File: test/e2e/specs/editor/various/sidebar-permalink.spec.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+			.getByRole( 'textbox', { name: 'Add title' } )
+			.fill( 'aaaaa' );
+		await editor.saveDraft();
+		// Start editing again.
+		await editor.canvas
+			.getByRole( 'textbox', { name: 'Add title' } )
+			.fill( 'aaaa (Updated)' );
+		await expect(
+			page.getByRole( 'button', { name: 'Change URL' } )
+		).toBeHidden();
```
Comment (WunderBart) [inline test/e2e/specs/editor/various/sidebar-permalink.spec.js:58]: As above!

File: test/e2e/specs/editor/various/sidebar-permalink.spec.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+			.fill( 'aaaaa' );
+		await editor.publishPost();
+
+		// Start editing again.
+		await editor.canvas
+			.getByRole( 'textbox', { name: 'Add title' } )
+			.fill( 'aaaa (Updated)' );
+		await expect(
+			page.getByRole( 'button', { name: 'Change URL' } )
+		).toBeVisible();
```
Comment (WunderBart) [inline test/e2e/specs/editor/various/sidebar-permalink.spec.js:79]: As above, but the other way round 😄

**Review by WunderBart — APPROVED** [review]
Looks good, just a few non-blocking comments. Thanks! 🚀

File: test/e2e/specs/editor/various/sidebar-permalink.spec.js (hunk 2 above)
Comment (Mamaduka) [reply test/e2e/specs/editor/various/sidebar-permalink.spec.js:38]: >Should we ensure this button is visible before the post-publish editing?

I don't think we need to do that.

The button should be rendered at all. Not sure why publishing step was included in the first place. Probably testing some regression.

**Pull request merged**